transport/src/main/java/io/netty/channel/ChannelOutboundInvokerCallback.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+/**
+ * Listener which will be notified when an {@link ChannelOutboundInvoker} methods is completed.
+ */
+// TODO: Should ChannelOutboundInvokerCallback be EventExecutor aware and so be able to delegate if needed ?
+public interface ChannelOutboundInvokerCallback {
+
+    // TODO: Should this be here and should we even support cancellation at all ?
+    static boolean setUncancellable(ChannelOutboundInvokerCallback callback) {
+        if (callback instanceof ChannelPromise) {
+            return ((ChannelPromise) callback).setUncancellable();
+        }
+        return true;
+    }
+
+    /**
+     * Called when an {@link ChannelOutboundInvoker} methods completes successfully.
+     */
+    void onSuccess();
+
+     /**
+     * Called when an {@link ChannelOutboundInvoker} methods completes non-successfully.
+      *
+     * @param cause the {@link Throwable} that caused the error.
+     */
+    void onError(Throwable cause);
+
+    /**
+     * A noop instance of an {@link ChannelOutboundInvokerCallback}. This can be used if you are not interested
+     * in the result of an {@link ChannelOutboundInvoker} at all.
+     *
+     * @return a noop instance.
+     */
+    static ChannelOutboundInvokerCallback noop() {
```

Comment:
If we imagine this in a static import, it wouldn't be very descriptive. Maybe calling this `noCallback` is better?

Reply:
@ejona86 i agree and think we should add something to the context etc that will call fireExceptionCaught. Maybe it would be even good to just remove the noop() as you suggested